glom/mutable.py

```diff
@@ -105,39 +117,59 @@ def __init__(self, path, val):
             self.op, self.arg = path.items()[-1]
         except IndexError:
             raise ValueError('path must have at least one element')
+        self._orig_path = path
         self.path = path[:-1]
 
         if self.op not in '[.P':
             # maybe if we add null-coalescing this should do something?
             raise ValueError('last part of path must be setattr or setitem')
         self.val = val
 
+        if missing is not None:
+            if not callable(missing):
+                raise TypeError('expected missing to be callable, not %r' % (missing,))
+        self.missing = missing
+
     def glomit(self, target, scope):
         if type(self.val) is Spec:
             val = scope[glom](target, self.val, scope)
         else:
             val = self.val
-        dest = scope[glom](target, self.path, scope)
+
+        op, arg, path = self.op, self.arg, self.path
+        try:
+            dest = scope[glom](target, self.path, scope)
+        except PathAccessError as pae:
+            if not self.missing:
+                raise
+
+            remaining_path = self._orig_path[pae.part_idx + 1:]
+            val = assign(self.missing(), remaining_path, val, missing=self.missing)
```

Comment:
we can get scope etc passing through with a small tweak

```scope[glom](target, Assign(remaining_path, missing, missing=self.missing), scope)```

Reply:
close! but the actual line was: `val = scope[glom](self.missing(), Assign(remaining_path, val, missing=self.missing), scope)` ;)